internals/src/slice.rs

```diff
@@ -24,6 +24,25 @@ pub trait SliceExt {
     /// let fail = slice.as_chunks_mut::<0>();
     /// ```
     fn bitcoin_as_chunks_mut<const N: usize>(&mut self) -> (&mut [[Self::Item; N]], &mut [Self::Item]);
+
+    /// Tries to access a sub-array of length `ARRAY_LEN` at the specified `offset`.
+    ///
+    /// Returns `None` in case of out-of-bounds access.
```

Comment:
Why not return an Error type with useful information instead of just `None`?  Although that can be done later not needed with this PR.

Reply:
Because these are low-level primitives which don't have much information about the error, so `ok_or` is likely better but also this one is currently only used to implement APIs consistent with `std` which return `Option` so any such information would be a waste.

Reply:
I see.  So besides mirroring std, since there is only one specific error then it's not informative to return why it error-ed it sounds like.  Thanks, just looking to understand better the rational for when to return None and when to return an Error.

Reply:
One is in iterator - trying to go past the end means end of iteration. The other is `get` method which is similar enough to `get` on slices to warrant matching API. `std` has `Option` there not `Result`. While returning indices in error types is somewhat convenient in erroneous cases using `get` is usually an operation that's handled by the caller.